dht/range_streamer.cc

```diff
@@ -291,8 +291,8 @@ future<> range_streamer::stream_async() {
                     for (auto it = range_vec.begin(); it < range_vec.end();) {
                         ranges_to_stream.push_back(*it);
                         ++it;
-                        auto percentage = _db.local().get_config().stream_plan_ranges_percentage();
-                        size_t nr_ranges_per_stream_plan = nr_ranges_total * percentage;
+                        auto fraction = _db.local().get_config().stream_plan_ranges_fraction();
+                        size_t nr_ranges_per_stream_plan = nr_ranges_total * fraction;
```

Comment:
but we still accept terribly wrong settings like `-100` and `99`. would be great to prevent scylla from starting with these options.

does it make sense to add a customized validator to `named_value`. so that we can add a, for instance, range checker. and the `allowed_values` is but a special case of the customized validator.

Reply:
Yes, I think it makes sense to add a special "fraction" value type, which is a float that has to be between 0 and 1. We already have several of these "fraction" configuration option. We need to make sure that the validation is done no matter where the value comes from (command line or scylla.yaml - or possibly CQL?).